src/common/types.cpp

```diff
@@ -1523,20 +1578,68 @@ shared_ptr<ExtraTypeInfo> ExtraTypeInfo::Deserialize(FieldReader &reader) {
 		extra_info = UserTypeInfo::Deserialize(reader);
 		break;
 	case ExtraTypeInfoType::ENUM_TYPE_INFO: {
-		auto enum_size = reader.ReadRequired<uint32_t>();
-		auto enum_internal_type = EnumVectorDictType(enum_size);
-		switch (enum_internal_type) {
-		case PhysicalType::UINT8:
-			extra_info = EnumTypeInfoTemplated<uint8_t>::Deserialize(reader, enum_size);
-			break;
-		case PhysicalType::UINT16:
-			extra_info = EnumTypeInfoTemplated<uint16_t>::Deserialize(reader, enum_size);
-			break;
-		case PhysicalType::UINT32:
-			extra_info = EnumTypeInfoTemplated<uint32_t>::Deserialize(reader, enum_size);
+		auto schema_name = reader.ReadRequired<string>();
+		auto enum_name = reader.ReadRequired<string>();
+		auto deserialize_internals = reader.ReadRequired<bool>();
+		if (!deserialize_internals) {
+			// this means the enum should already be in the catalog.
+			auto &client_context = reader.GetSource().GetContext();
+			// See if the serializer has a catalog
+			auto catalog = reader.GetSource().GetCatalog();
+			if (catalog) {
+				auto enum_type = catalog->GetType(client_context, schema_name, enum_name, true);
+				if (enum_type != LogicalType::INVALID) {
+					extra_info = enum_type.GetAuxInfoShrPtr();
+					break;
+				}
+			}
+			// Try the DB Path as a catalog name
+			auto catalog_name = AttachedDatabase::ExtractDatabaseName(client_context.db->config.options.database_path);
+			if (Catalog::TypeExists(client_context, catalog_name, schema_name, enum_name)) {
```

Comment:
The logic here seems to duplicate the standard catalog resolution logic - can we just make this `INVALID_CATALOG` instead and let the catalog handle where to look up the enum?

Also - is all of this necessary in the first place? Shouldn't there just always be a catalog that we are deserializing from?

Reply:
Not always. At the ` Planner::VerifyPlan`, there is no catalog, for example.
Indeed using `INVALID_CATALOG` does the trick there :-)